src/internal/m365/service/exchange/export_test.go

```diff
@@ -0,0 +1,433 @@
+						Path: p,
+						ItemData: []data.Item{
+							&dataMock.Item{
+								ItemID:  "id1",
+								ReadErr: assert.AnError,
+							},
+							&dataMock.Item{
+								ItemID: "id2",
+								Reader: io.NopCloser(bytes.NewReader(emailBodyBytes)),
+							},
+						},
+					},
+				},
+			},
+			expectedItems: [][]export.Item{
+				{
+					{
+						ID:   "id2",
+						Name: "id2.eml",
+						Body: io.NopCloser(bytes.NewReader(emailBodyBytes)),
+					},
+					{
+						ID:    "",
+						Error: assert.AnError,
+					},
+				},
+			},
+		},
+		{
+			name: "multiple collections",
+			dcs: []data.RestoreCollection{
+				data.FetchRestoreCollection{
+					Collection: dataMock.Collection{
+						Path: p,
+						ItemData: []data.Item{
+							&dataMock.Item{
+								ItemID: "id1",
+								Reader: io.NopCloser(bytes.NewReader(emailBodyBytes)),
+							},
+						},
+					},
+				},
+				data.FetchRestoreCollection{
+					Collection: dataMock.Collection{
+						Path: p,
+						ItemData: []data.Item{
+							&dataMock.Item{
+								ItemID: "id2",
+								Reader: io.NopCloser(bytes.NewReader(emailBodyBytes)),
+							},
+						},
+					},
+				},
+			},
+			expectedItems: [][]export.Item{
+				{
+					{
+						ID:   "id1",
+						Name: "id1.eml",
+						Body: io.NopCloser(bytes.NewReader(emailBodyBytes)),
+					},
+				},
+				{
+					{
+						ID:   "id2",
+						Name: "id2.eml",
+						Body: io.NopCloser(bytes.NewReader(emailBodyBytes)),
+					},
+				},
+			},
+		},
+		{
+			name: "collection without exchange category",
+			dcs: []data.RestoreCollection{
+				data.FetchRestoreCollection{
+					Collection: dataMock.Collection{
+						Path: p2,
+						ItemData: []data.Item{
+							&dataMock.Item{
+								ItemID: "id1",
+								Reader: io.NopCloser(bytes.NewReader(emailBodyBytes)),
+							},
+						},
+					},
+				},
+			},
+			expectedItems: [][]export.Item{},
+			hasErr:        true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			exportCfg := control.ExportConfig{}
+			stats := data.ExportStats{}
+
+			ecs, err := NewExchangeHandler(control.DefaultOptions()).
+				ProduceExportCollections(
+					ctx,
+					int(version.Backup),
+					exportCfg,
+					tt.dcs,
+					&stats,
+					fault.New(true))
+
+			if tt.hasErr {
+				assert.Error(t, err, "export collections error")
+				return
+			}
+
+			assert.NoError(t, err, "export collections error")
+			assert.Len(t, ecs, len(tt.expectedItems), "num of collections")
+
+			expectedStats := data.ExportStats{}
+
+			for c := range ecs {
+				i := -1
+				for item := range ecs[c].Items(ctx) {
+					i++
+
+					size := 0
+
+					if item.Body == nil {
+						assert.ErrorIs(t, item.Error, tt.expectedItems[c][i].Error)
+						continue
+					}
+
+					// unwrap the body from stats reader
+					b, err := io.ReadAll(item.Body)
+					assert.NoError(t, err, clues.ToCore(err))
+
+					size += len(b)
+					bitem := io.NopCloser(bytes.NewBuffer(b))
+					item.Body = bitem
```

Comment:
same deal here, do we need to reset the reader here? It doesn't seem like other parts of the test use it?